### PR TITLE
Visual filter for satellites

### PR DESCRIPTION
**Pull request by alex-w**
### Description
This pull request has 2 serious changes:
- Removing the 'altitude' option from 'custom filter' of satellites;
- Adding visual filter for satellites.

The visual filter has 3 options and when it activated all visible satellites will hiding outside valid range of altitudes. This visual filter suppress the rendering of orbit lines - this is important!

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (header file)
- [x] I have updated the respective chapter in the Stellarium User Guide
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by alex-w** [comment]
OK, it's ready for merge

File: plugins/Satellites/src/Satellites.hpp [hunk 1]
```diff
@@ -663,20 +665,17 @@ public slots:
...
-	bool getFlagCFAltitude() { return Satellite::flagCFAltitude; }
+	//! Set whether visual filter 'altitude' enabled.
+	void setFlagVFAltitude(bool b);
+	bool getFlagVFAltitude() { return Satellite::flagVFAltitude; }
 
-	//! Set whether custom filter 'altitude' maximum value (in kilometers).
-	//! Emits customFilterChanged()
-	void setMaxCFAltitude(double v);
-	double getMaxCFAltitude() { return Satellite::maxCFAltitude; }
+	//! Set whether visual filter 'altitude' maximum value (in kilometers).
```
Comment (gzotti) [inline plugins/Satellites/src/Satellites.hpp:672]: delete "whether"

File: plugins/Satellites/src/Satellites.hpp [hunk 2]
```diff
@@ -663,20 +665,17 @@ public slots:
...
+	void setMaxVFAltitude(double v);
+	double getMaxVFAltitude() { return Satellite::maxVFAltitude; }
 
-	//! Set whether custom filter 'altitude' minimum value (in kilometers).
-	//! Emits customFilterChanged()
-	void setMinCFAltitude(double v);
-	double getMinCFAltitude() { return Satellite::minCFAltitude; }
+	//! Set whether visual filter 'altitude' minimum value (in kilometers).
```
Comment (gzotti) [inline plugins/Satellites/src/Satellites.hpp:676]: delete "whether"

File: plugins/Satellites/src/gui/SatellitesDialog.cpp [hunk 3]
```diff
@@ -311,6 +317,12 @@ void SatellitesDialog::createDialogContent()
 	ui->validAgeLabel->setStyleSheet(style);
 }
 
+void SatellitesDialog::updateMinMaxAltitude(bool state)
```
Comment (gzotti) [inline plugins/Satellites/src/gui/SatellitesDialog.cpp:320]: better name: enableMinMaxAltitude

File: plugins/Satellites/src/gui/SatellitesDialog.cpp [hunk 4]
```diff
@@ -1137,11 +1150,16 @@ void SatellitesDialog::populateInfo()
 	ui->updateFrequencySpinBox->setSuffix(QString(" %1").arg(qc_("h","time")));
 	// TRANSLATORS: Unit of measure for distance - kilometers
 	QString km = qc_("km", "distance");
+	ui->minAltitude->setSuffix(QString(" %1").arg(km));
+	ui->minAltitude->setToolTip(QString("%1 %2..%3 %4").arg(vr, QString::number(ui->minAltitude->minimum(), 'f', 0), QString::number(ui->minAltitude->maximum(), 'f', 0), km));
+	ui->maxAltitude->setSuffix(QString(" %1").arg(km));
+	ui->maxAltitude->setToolTip(QString("%1 %2..%3 %4").arg(vr, QString::number(ui->maxAltitude->minimum(), 'f', 0), QString::number(ui->maxAltitude->maximum(), 'f', 0), km));
+	ui->altitudeCheckBox->setToolTip(QString("<p>%1</p>").arg(q_("Display satellites from range of altitudes only. This option suppress rendering orbit lines!")));
```
Comment (gzotti) [inline plugins/Satellites/src/gui/SatellitesDialog.cpp:1157]: "Display satellites within selected range of altitudes only. This option suppresses rendering orbit lines!"

File: plugins/Satellites/src/gui/SatellitesDialog.cpp [hunk 5]
```diff
@@ -1137,11 +1150,16 @@ void SatellitesDialog::populateInfo()
...
-	ui->orbitDurationSpin->setToolTip(QString("<p>%1. %2: %3-%4%5</p>").arg(q_("Segment length: duration of a single segment in seconds"), q_("Valid range"), QString::number(ui->orbitDurationSpin->minimum()), QString::number(ui->orbitDurationSpin->maximum()), s));
-	ui->orbitFadeSpin->setToolTip(QString("<p>%1. %2: %3-%4</p>").arg(q_("Fade length: number of segments used to draw each end of the line"), q_("Valid range"), QString::number(ui->orbitFadeSpin->minimum()), QString::number(ui->orbitFadeSpin->maximum())));
+	ui->umbraDistance->setToolTip(QString("<p>%1. %2 %3..%4 %5</p>").arg(q_("Distance to the center of umbra from Earth's surface (height of imagined satellite)"), vr, QString::number(ui->umbraDistance->minimum(), 'f', 1), QString::number(ui->umbraDistance->maximum(), 'f', 1), km));
+	ui->orbitSegmentsSpin->setToolTip(QString("<p>%1. %2 %3..%4</p>").arg(q_("Number of  segments: number of segments used to draw the line"), vr, QString::number(ui->orbitSegmentsSpin->minimum()), QString::number(ui->orbitSegmentsSpin->maximum())));
```
Comment (gzotti) [inline plugins/Satellites/src/gui/SatellitesDialog.cpp:1160]: Remove excess space in "Number of _ _ segments"

**Comment by alex-w** [comment]
OK, done

File: plugins/Satellites/src/gui/SatellitesDialog.cpp [hunk 6]
```diff
@@ -1121,11 +1134,16 @@ void SatellitesDialog::populateInfo()
 	ui->updateFrequencySpinBox->setSuffix(QString(" %1").arg(qc_("h","time")));
 	// TRANSLATORS: Unit of measure for distance - kilometers
 	QString km = qc_("km", "distance");
+	ui->minAltitude->setSuffix(QString(" %1").arg(km));
+	ui->minAltitude->setToolTip(QString("%1 %2..%3 %4").arg(vr, QString::number(ui->minAltitude->minimum(), 'f', 0), QString::number(ui->minAltitude->maximum(), 'f', 0), km));
+	ui->maxAltitude->setSuffix(QString(" %1").arg(km));
+	ui->maxAltitude->setToolTip(QString("%1 %2..%3 %4").arg(vr, QString::number(ui->maxAltitude->minimum(), 'f', 0), QString::number(ui->maxAltitude->maximum(), 'f', 0), km));
+	ui->altitudeCheckBox->setToolTip(QString("<p>%1</p>").arg(q_("Display satellites within selected range of altitudes only. This option suppresses rendering orbit lines!")));
```
Comment (gzotti) [inline plugins/Satellites/src/gui/SatellitesDialog.cpp:1141]: Sorry. "...suppresses rendering _of_ orbit lines"

Comment (alex-w) [reply plugins/Satellites/src/gui/SatellitesDialog.cpp]: Fixed

File: plugins/Satellites/src/Satellites.hpp (hunk 1 above)
Comment (alex-w) [reply plugins/Satellites/src/Satellites.hpp]: Fixed

File: plugins/Satellites/src/Satellites.hpp (hunk 2 above)
Comment (alex-w) [reply plugins/Satellites/src/Satellites.hpp]: Fixed

File: plugins/Satellites/src/gui/SatellitesDialog.hpp [hunk 7]
```diff
@@ -138,6 +138,8 @@ private slots:
 	void setRightSideToROMode();
 	void setRightSideToRWMode();
 
+	void enableMinMaxAltitude(bool state);
+
 private:
 	//! @todo find out if this is really necessary... --BM
```
Comment (gzotti) [inline plugins/Satellites/src/gui/SatellitesDialog.hpp:144]: Any resolution here? (Not strictly part of the PR, but this question is over 9 years old now...)

File: plugins/Satellites/src/gui/SatellitesDialog.cpp (hunk 4 above)
Comment (alex-w) [reply plugins/Satellites/src/gui/SatellitesDialog.cpp]: Fixed

File: plugins/Satellites/src/gui/SatellitesDialog.cpp (hunk 5 above)
Comment (alex-w) [reply plugins/Satellites/src/gui/SatellitesDialog.cpp]: Fixed

File: plugins/Satellites/src/gui/satellitesDialog.ui [hunk 8]
```diff
@@ -577,56 +589,31 @@
...
-               <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
-              </property>
-              <property name="decimals">
-               <number>1</number>
-              </property>
-              <property name="minimum">
-               <double>75.000000000000000</double>
+             <widget class="QToolButton" name="transitColorButton">
+              <property name="toolTip">
+               <string>Color of markers for satellites in transit through the Sun or the Moon</string>
```
[the remark this answers is not in the document]
Comment (alex-w) [reply plugins/Satellites/src/gui/satellitesDialog.ui]: Fixed

File: plugins/Satellites/src/gui/SatellitesDialog.hpp (hunk 7 above)
Comment (alex-w) [reply plugins/Satellites/src/gui/SatellitesDialog.hpp:144]: I didn’t checked Bogdan’s comment, maybe situation is gone already

**Review by gzotti — APPROVED** [review]
OK, I don't see further issues.

**Comment by alex-w** [comment]
> OK, I don't see further issues.

Thanks!

**Pull request merged**